File: 2015/10/javascript/main.mjs
import fs from "node:fs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Functions ---
function lookAndSay(s) {
	const a = Array.from(s);
	let res = [];
	let last = a[0];
	let count = 1;
	for (let i = 1; i < a.length; i++) {
		if (a[i] === last) count++;
		else {
			res.push(Array.from(count.toString()));
			res.push(last);
			count = 1;
		}
		last = a[i];
	}
	res.push(Array.from(count.toString()));
	res.push(last);
	return res.join("");
}

// --- Parse the input file ---
let text = fs.readFileSync(INPUT_FILE, 'utf8').trim();

// --- Puzzle 1 ---
for (let i = 0; i < 40; i++) text = lookAndSay(text);
console.log("1. Length of the result: ", text.length.toLocaleString());

// --- Puzzle 2 ---
for (let i = 0; i < 10; i++) text = lookAndSay(text);
console.log("2. Length of the result: ", text.length.toLocaleString());

File: 2015/10/go/main.go
package main

import (
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

// --- Constants ---
const FILENAME = "../input/input.txt"

func lookAndSay(s string) string {
	if len(s) == 0 {
		return ""
	}

	var res strings.Builder
	a := []byte(s)
	last := a[0]
	count := 1

	for i := 1; i < len(a); i++ {
		if a[i] == last {
			count++
		} else {
			res.WriteString(strconv.Itoa(count))
			res.WriteByte(last)
			last = a[i]
			count = 1
		}
	}
	res.WriteString(strconv.Itoa(count))
	res.WriteByte(last)
	return res.String()
}

func main() {
	// --- Read input file ---
	bytes, err := os.ReadFile(FILENAME)
	if err != nil {
		log.Fatal(err)
	}
	text := strings.TrimSpace(string(bytes))

	// --- Puzzle 1 ---
	for i := 0; i < 40; i++ {
		text = lookAndSay(text)
	}
	fmt.Println("1. Length of the result:", len(text))

	// --- Puzzle 2 ---
	for i := 0; i < 10; i++ {
		text = lookAndSay(text)
	}
	fmt.Println("2. Length of the result:", len(text))
}
